citations/citations-impl/impl/src/java/org/sakaiproject/citation/impl/DbCitationService.java

```diff
@@ -546,6 +556,47 @@ protected CitationCollection getUnnestedCitationColl(String collectionId)
 		}
 
 		/* (non-Javadoc)
+		* @see org.sakaiproject.citation.impl.BaseCitationService.Storage#getNextCitCollectionOrderIdForCollection(java.lang.String)
+		*/
+		protected String getNextCitCollectionOrderIdForCollection(String collectionId)
+		{
+			String statement = "SELECT MAX(LOCATION)+1 FROM " + m_collectionOrderTableName + " where COLLECTION_ID = ? AND SECTION_TYPE IS NOT NULL ";
```

Comment:
Doesn't this have problems with multiple transactions in that one the transaction is finished the max(location) might be different?